blockchain.js

```diff
@@ -0,0 +1,626 @@
+  const seed = await bip39.mnemonicToSeed(mnemonic);
+  return flow.crypto.genKeys({
+    entropy: seed.toString('hex'),
+    entropyEnc: 'hex',
+  });
+};
+
+function uint8Array2hex(uint8Array) {
+  return Array.prototype.map.call(uint8Array, x => ('00' + x.toString(16)).slice(-2)).join('');
+}
+const _isSealed = tx => tx.status >= 4;
+const _waitForTx = async txid => {
+  for (;;) {
+    const response2 = await flow.sdk.send(await flow.sdk.pipe(await flow.sdk.build([
+      flow.sdk.getTransactionStatus(txid),
+    ]), [
+      flow.sdk.resolve([
+        flow.sdk.resolveParams,
+      ]),
+    ]), { node: flowConstants.host });
+    // console.log('got response 2', response2);
+    if (_isSealed(response2.transaction)) {
+      return response2;
+    } else {
+      await new Promise((accept, reject) => {
+        setTimeout(accept, 500);
+      });
+    }
+  }
+};
+const _createAccount = async () => {
+  const res = await fetch(accountsHost, {
+    method: 'POST',
+  });
+  const j = await res.json();
+  return j;
+};
+const _bakeContract = async (contractKeys, contractSource) => {
+  const acctResponse = await flow.sdk.send(await flow.sdk.pipe(await flow.sdk.build([
+    flow.sdk.getAccount(contractKeys.address),
+  ]), [
+    flow.sdk.resolve([
+      flow.sdk.resolveParams,
+    ]),
+  ]), { node: flowConstants.host });
+  const seqNum = acctResponse.account.keys[0].sequenceNumber;
+
+  const signingFunction = flow.signingFunction.signingFunction(contractKeys.privateKey);
+
+  const response = await flow.sdk.send(await flow.sdk.pipe(await flow.sdk.build([
+    flow.sdk.authorizations([flow.sdk.authorization(contractKeys.address, signingFunction, 0)]),
+    flow.sdk.payer(flow.sdk.authorization(contractKeys.address, signingFunction, 0)),
+    flow.sdk.proposer(flow.sdk.authorization(contractKeys.address, signingFunction, 0, seqNum)),
+    flow.sdk.limit(100),
+    flow.sdk.transaction`
+      transaction(code: String) {
+        prepare(acct: AuthAccount) {
+          acct.setCode(code.decodeHex())
+        }
+      }
+    `,
+  ]), [
+    flow.sdk.args([
+      flow.sdk.arg(uint8Array2hex(new TextEncoder().encode(contractSource)), flow.types.String),
+    ]),
+    flow.sdk.resolve([
+      flow.sdk.resolveArguments,
+      flow.sdk.resolveParams,
+      flow.sdk.resolveAccounts,
+      flow.sdk.resolveRefBlockId({ node: flowConstants.host }),
+      flow.sdk.resolveSignatures,
+    ]),
+  ]), { node: flowConstants.host });
+ 
+  const response2 = await _waitForTx(response.transactionId);
+  return response2;
+};
+const _bakeUserAccount = async (userKeys, ftContractAddress, nftContractAddress) => {
+  // set up ft
+  {
+    const acctResponse = await flow.sdk.send(await flow.sdk.pipe(await flow.sdk.build([
+      flow.sdk.getAccount(userKeys.address),
+    ]), [
+      flow.sdk.resolve([
+        flow.sdk.resolveParams,
+      ]),
+    ]), { node: flowConstants.host });
+    const seqNum = acctResponse.account.keys[0].sequenceNumber;
```

Comment:
This is duplicated 8 times in the code, maybe break this out as it's own function if possible.

Reply:
Yes, that's a good call